spec/buffered_line_reader_spec.rb

```diff
@@ -74,4 +74,24 @@ def tests_for_terminator(term, desc)
       "fourth line", "", "last"]
     expect(subject.lines_from(chunks).to_a).to eq(expected)
   end
+
+  it "decodes from UTF-8" do
+    text = "abc€豆腐xyz"
+    chunks = [(text + "\n").encode("UTF-8").b]
+    expected = [text]
+    expect(subject.lines_from(chunks).to_a).to eq(expected)
+  end
+
+  it "decodes from UTF-8 when multi-byte characters are split across chunks" do
```

Comment:
I verified that this test fails without the fix. The equivalent test in the SSE contract tests also fails without the fix and passes with the fix.